pkg/network/ebpf/c/protocols/protocol-classification-helpers.h

```diff
@@ -103,11 +108,22 @@ static __always_inline bool is_http(const char *buf, __u32 size) {
 struct pg_message_header {
     __u8 message_tag;
     __u32 message_len; // Big-endian: use bpf_ntohl to read this field
-} __attribute__((aligned(8)));
-#define PG_HDR_SIZE 5
+} __attribute__((packed));
 
-static __always_inline bool is_postgres(const char *buf, __u32 buf_size) {
-    CHECK_PRELIMINARY_BUFFER_CONDITIONS(buf, buf_size, POSTGRES_MIN_MSG_SIZE);
+static __always_inline bool is_postgres_connect(const char *buf, __u32 buf_size) {
+    CHECK_PRELIMINARY_BUFFER_CONDITIONS(buf, buf_size, POSTGRES_STARTUP_MIN_LEN);
+
+    struct pg_startup_header *hdr = (struct pg_startup_header *)buf;
+
+    if (bpf_ntohl(hdr->version) != PG_STARTUP_VERSION) {
+        return false;
+    }
+
+    return !bpf_memcmp(buf + sizeof(*hdr), PG_STARTUP_USER_PARAM, sizeof(PG_STARTUP_USER_PARAM));
```

Comment:
```suggestion
    return !bpf_memcmp(buf + sizeof(*hdr), PG_STARTUP_USER_PARAM, sizeof(PG_STARTUP_USER_PARAM)-1);
```

Reply:
Actually, the Postgres protocol uses C-style strings, so I think it is good idea to check for the null byte at the end as well.

Reply:
so let's document that